common/lib/client/auth.js

```diff
@@ -97,7 +108,33 @@ var Auth = (function() {
 			Logger.logAction(Logger.LOG_ERROR, 'Auth()', msg);
 			throw new Error(msg);
 		}
-	}
+	};
+
+	/**
+	 * Update the authentication options for this client;
+	 * this will also update these options on the realtime connection
+	 * so that they take effect immediately
+	 * @param options
+	 */
+	Auth.prototype.setOptions = function(options) {
+		this._setOptions(options);
+		if(this.client instanceof Realtime) {
+			this.client.connection.connectionManager.onAuthUpdated();
+		}
+	};
+
+	Auth.prototype.setClientId = function(clientId) {
+		if(clientId !== '*') {
+			if(this.clientId && this.clientId !== clientId) {
+				var msg = 'Unable to set incompatible clientId';
+				return new ErrorInfo(msg, 40102, 401);
+			}
+			/* RSA7a4: if options.clientId is provided and is not
+			 * null, it overrides defaultTokenParams.clientId */
+			this.clientId = this.tokenParams.clientId = clientId;
+		}
```

Comment:
If someone tries to call this with clientId of '\*', we should probably throw an error ([as we do now](https://github.com/ably/ably-js/blob/e6319f9dc8d5ea802cdbfb6b24b09175231f7300/common/lib/client/rest.js#L39-L40) if you try and put `clientId: '*'` in clientOptions) rather than silently ignoring it, or people might thing that that's a valid way of unsetting a clientId
```
} else {
  return new ErrorInfo('Can’t use "*" as a clientId as that string is reserved. (To change the default token request behaviour to use a wildcard clientId, pass in {defaultTokenParams: {clientId: "*"}})', 40012, 400);
}
```